### ISSUE TITLE
Multiple patterns delimited by spaces

### ISSUE THREAD

**Issue by notramo** [jhawthorn/fzy#67]
Please add a mode to `fzy` so search with multiple patterns. (see the `EXTENDED SEARCH MODE` in `fzf` man page)
In this mode, spaces don't match a literal space character, but acts as delimiters of patterns, and each pattern is matched independently.
In this mode, with two patterns, a string should get lower score, if the two patterns matches directly next to each other.
A string should get a higher score, if the patterns match in the order they are given in the prompt.

**Comment by voidus** [comment]
Yes please. When selecting files in vim, I'll often notice that there are multiple matches in different directories. With fzf, it's very convenient to type a space and then a match for the directory.
